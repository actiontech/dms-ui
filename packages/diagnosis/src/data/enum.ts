export enum AdminRole {
  admin = 'admin'
}

export enum AdminUser {
  admin = 'admin'
}
