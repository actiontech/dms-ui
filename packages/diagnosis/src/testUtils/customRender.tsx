import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter, MemoryRouterProps } from 'react-router-dom';
import { Dictionary } from '@actiontech/shared/lib/types/common.type';
import { storeFactory } from './mockRedux';
import { StyledEngineProvider, ThemeProvider } from '@mui/system';
import { mount } from 'enzyme';
import { ConfigProvider } from 'antd';
import zhCN from 'antd/locale/zh_CN';

import lightTheme from '../theme/light';
import sharedTheme from '@actiontech/shared/lib/theme/light';

type MountParams = Parameters<typeof mount>;
type RenderParams = Parameters<typeof render>;

export const renderWithReduxAndTheme = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(ui, {
    wrapper: ({ children }) => (
      <ThemeProvider theme={{ ...lightTheme, ...sharedTheme }}>
        <Provider store={storeFactory(initStore)}>{children}</Provider>
      </ThemeProvider>
    ),
    ...Option
  });
};

export const renderWithTheme = (...[ui, option]: [...RenderParams]) => {
  return render(
    <ThemeProvider theme={{ ...lightTheme, ...sharedTheme }}>
      {ui}
    </ThemeProvider>,
    option
  );
};

export const renderWithThemeAndRouter = (
  ...[ui, option, props]: [...RenderParams, MemoryRouterProps?]
) => {
  return render(
    <MemoryRouter {...props}>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={{ ...lightTheme, ...sharedTheme }}>
          {ui}
        </ThemeProvider>
      </StyledEngineProvider>
    </MemoryRouter>,
    option
  );
};

export const superRender = (
  ...[ui, option, otherProps]: [
    ...RenderParams,
    {
      routerProps?: MemoryRouterProps;
      initStore?: any;
    }?
  ]
) => {
  const renderReturn = render(ui, {
    wrapper: ({ children }) => {
      return (
        <ConfigProvider locale={zhCN}>
          <ThemeProvider theme={{ ...lightTheme, ...sharedTheme }}>
            <MemoryRouter {...otherProps?.routerProps}>
              <Provider store={storeFactory(otherProps?.initStore)}>
                {children}
              </Provider>
            </MemoryRouter>
          </ThemeProvider>
        </ConfigProvider>
      );
    },
    ...option
  });
  return renderReturn;
};

export const mountWithTheme = (...[ui, option]: [...MountParams]) => {
  return mount(ui, {
    ...option,
    wrappingComponent: ThemeProvider,
    wrappingComponentProps: {
      theme: { ...lightTheme, ...sharedTheme }
    }
  });
};
