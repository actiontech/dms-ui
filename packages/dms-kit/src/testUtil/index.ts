export * from './superRender';
export * from './customQuery';
export * from './common';
