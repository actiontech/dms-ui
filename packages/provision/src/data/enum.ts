export enum ModalName {
  GetConnection = 'GET_CONNECTION',

  UpdateTemplateInAuth = 'UPDATE_TEMPLATE_IN_AUTH',

  UpdateUserInAuth = 'UPDATE_USER_IN_AUTH',

  UpdateExpirationInAuth = 'UPDATE_EXPIRATION_IN_AUTH',

  DataPermissionModal = 'DATA_PERMISSION_MODAL',

  CopyTemplate = 'COPY_TEMPLATE'
}

export enum EventEmitterKey {
  Refresh_Auth_List_Table = 'Refresh_Auth_List_Table',

  Refresh_Auth_Template_List_Table = 'Refresh_Auth_Template_List_Table'
}
