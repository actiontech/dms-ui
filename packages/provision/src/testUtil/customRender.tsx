import { ThemeProvider } from '@mui/system';
import {
  render,
  renderHook,
  RenderHookOptions,
  RenderHookResult
} from '@testing-library/react';
import {
  BrowserRouter,
  MemoryRouter,
  MemoryRouterProps
} from 'react-router-dom';
import { MutableSnapshot, RecoilRoot } from 'recoil';
import SyncRecoil from '~/utils/SyncRecoil';
import { JSXElementConstructor } from 'react';
import lightTheme from '@actiontech/shared/lib/theme/light';

type RenderParams = Parameters<typeof render>;
type RecoilRootPropsCustom = {
  initializeState?: (mutableSnapshot: MutableSnapshot) => void;
  override?: boolean;
};

export const renderWithRecoilRootWithRouter = (
  ...[ui, option, recoilRootProps]: [...RenderParams, RecoilRootPropsCustom?]
) => {
  return render(
    <BrowserRouter>
      <RecoilRoot {...recoilRootProps}>{ui}</RecoilRoot>
    </BrowserRouter>,
    option
  );
};

export const renderWithRecoilRootAndRouterAndTheme = (
  ...[ui, option, otherProps]: [
    ...RenderParams,
    {
      recoilRootProps?: RecoilRootPropsCustom;
    }?
  ]
) => {
  return render(
    <ThemeProvider theme={lightTheme}>
      <BrowserRouter>
        <RecoilRoot {...otherProps?.recoilRootProps}>{ui}</RecoilRoot>
      </BrowserRouter>
    </ThemeProvider>,
    option
  );
};

export const renderWithRecoilRootWithMemoryRouter = (
  ...[ui, option, otherProps]: [
    ...RenderParams,
    {
      routerProps?: MemoryRouterProps;
      recoilRootProps?: RecoilRootPropsCustom;
    }?
  ]
) => {
  return render(
    <MemoryRouter {...otherProps?.routerProps}>
      <RecoilRoot {...otherProps?.recoilRootProps}>{ui}</RecoilRoot>
    </MemoryRouter>,
    option
  );
};

export const renderWithRecoilRootAndMemoRouterAndTheme = (
  ...[ui, option, otherProps]: [
    ...RenderParams,
    {
      routerProps?: MemoryRouterProps;
      recoilRootProps?: RecoilRootPropsCustom;
    }?
  ]
) => {
  return render(
    <ThemeProvider theme={lightTheme}>
      <MemoryRouter {...otherProps?.routerProps}>
        <RecoilRoot {...otherProps?.recoilRootProps}>
          <SyncRecoil />
          {ui}
        </RecoilRoot>
      </MemoryRouter>
    </ThemeProvider>,
    option
  );
};

export const superRenderHooksWrapperFactory = (
  otherChildren?: React.ReactNode,
  recoilRootProps?: RecoilRootPropsCustom
): JSXElementConstructor<any> => {
  return ({ children }: { children: React.ReactNode }) => (
    <RecoilRoot {...recoilRootProps}>
      {otherChildren}
      {children}
      <SyncRecoil />
    </RecoilRoot>
  );
};

export const superRenderHooks = <Result, Props>(
  render: (initialProps: Props) => Result,
  options?: RenderHookOptions<Props> & {
    otherChildren?: React.ReactNode;
    recoilRootProps?: RecoilRootPropsCustom;
  }
): RenderHookResult<Result, Props> => {
  return renderHook(render, {
    wrapper: superRenderHooksWrapperFactory(
      options?.otherChildren,
      options?.recoilRootProps
    ),
    ...options
  });
};
