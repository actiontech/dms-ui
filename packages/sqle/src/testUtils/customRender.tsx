import {
  RenderResult,
  render,
  renderHook as renderHookByReact
} from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import {
  BrowserRouter,
  Router,
  RouterProps,
  MemoryRouter,
  MemoryRouterProps,
  useLocation
} from 'react-router-dom';
import { Provider } from 'react-redux';
import { storeFactory } from './mockRedux';
import { ConfigProvider, theme as antdTheme } from 'antd';
import { ThemeProvider, StyledEngineProvider } from '@mui/material/styles';
import { mount, shallow } from 'enzyme';
import zhCN from 'antd/locale/zh_CN';
import { Dictionary } from '@actiontech/shared/lib/types/common.type';

import sharedTheme from '@actiontech/shared/lib/theme/light';
import lightTheme from '../theme/light';

type RenderParams = Parameters<typeof render>;
type MountParams = Parameters<typeof mount>;
type ShallowParams = Parameters<typeof shallow>;

const themeData = {
  ...sharedTheme,
  ...lightTheme
};

export const renderWithRouter = (...[ui, option]: [...RenderParams]) => {
  return render(<BrowserRouter>{ui}</BrowserRouter>, option);
};

export const renderWithRouterAndRedux = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <BrowserRouter>
      <Provider store={storeFactory(initStore)}>{ui}</Provider>
    </BrowserRouter>,
    option
  );
};

export const renderWithRedux = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <Provider store={storeFactory(initStore)}>{ui}</Provider>,
    option
  );
};

export const renderHooksWithRedux = <TProps, TResult>(
  hooks: (props: TProps) => TResult,
  storeState: Dictionary = {}
) => {
  return renderHook(hooks, {
    wrapper: Provider,
    initialProps: {
      store: storeFactory(storeState)
    }
  } as any);
};

export const renderHooksWithTheme = <TProps, TResult>(
  hooks: (props: TProps) => TResult
) => {
  return renderHookByReact(hooks, {
    wrapper: ({ children }) => (
      <ConfigProvider
        locale={zhCN}
        theme={{ algorithm: antdTheme.defaultAlgorithm }}
      >
        <StyledEngineProvider injectFirst>
          <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
        </StyledEngineProvider>
      </ConfigProvider>
    )
  });
};

export const renderWithTheme = (...[ui, option]: [...RenderParams]) => {
  return render(
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={themeData}>{ui}</ThemeProvider>
    </StyledEngineProvider>,
    option
  );
};

export const renderWithThemeAndRedux = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <Provider store={storeFactory(initStore)}>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={themeData}>{ui}</ThemeProvider>
      </StyledEngineProvider>
    </Provider>,
    option
  );
};

export const renderWithThemeAndServerRouter = (
  ...[ui, option, props]: [...RenderParams, RouterProps]
) => {
  return render(
    <Router {...props}>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={themeData}>{ui}</ThemeProvider>
      </StyledEngineProvider>
    </Router>,
    option
  );
};

export const renderWithThemeAndRouter = (
  ...[ui, option, props]: [...RenderParams, MemoryRouterProps?]
) => {
  return render(
    <MemoryRouter {...props}>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={themeData}>{ui}</ThemeProvider>
      </StyledEngineProvider>
    </MemoryRouter>,
    option
  );
};

export const superRender = (
  ...[ui, option, otherProps]: [
    ...RenderParams,
    {
      routerProps?: MemoryRouterProps;
      initStore?: any;
    }?
  ]
) => {
  const renderReturn = render(ui, {
    wrapper: ({ children }) => {
      return (
        <Provider store={storeFactory(otherProps?.initStore)}>
          <MemoryRouter {...otherProps?.routerProps}>
            <StyledEngineProvider injectFirst>
              <ThemeProvider theme={themeData}>{children}</ThemeProvider>
            </StyledEngineProvider>
          </MemoryRouter>
        </Provider>
      );
    },
    ...option
  });
  return renderReturn;
};

export const mountWithTheme = (...[ui, option]: [...MountParams]) => {
  return mount(ui, {
    ...option,
    wrappingComponent: ThemeProvider,
    wrappingComponentProps: {
      theme: themeData
    }
  });
};

export const shallowWithRouter = (...[ui, option]: [...ShallowParams]) => {
  return shallow(ui, {
    ...option,
    wrappingComponent: MemoryRouter
  });
};

export const renderLocationDisplay = (): [
  () => RenderResult,
  () => JSX.Element
] => {
  const LocationComponent = () => {
    const location = useLocation();

    return <div data-testid="location-display">{location.pathname}</div>;
  };

  return [() => renderWithRouter(<LocationComponent />), LocationComponent];
};

export const superRender = (
  ...[ui, option, otherProps]: [
    ...RenderParams,
    {
      routerProps?: MemoryRouterProps;
      initStore?: any;
    }?
  ]
) => {
  const renderReturn = render(ui, {
    wrapper: ({ children }) => {
      return (
        <Provider store={storeFactory(otherProps?.initStore)}>
          <MemoryRouter {...otherProps?.routerProps}>
            <ThemeProvider theme={themeData}>{children}</ThemeProvider>
          </MemoryRouter>
        </Provider>
      );
    },
    ...option
  });
  return renderReturn;
};
