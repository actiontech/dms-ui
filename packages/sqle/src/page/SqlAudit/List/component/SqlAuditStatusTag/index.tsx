import { useTranslation } from 'react-i18next';
import { getSQLAuditRecordsV1FilterSqlAuditStatusEnum } from '@actiontech/shared/lib/api/sqle/service/sql_audit_record/index.enum';
import { useMemo } from 'react';
import { BasicTag } from '@actiontech/shared';
import { BasicTagColor } from '@actiontech/shared/lib/theme/theme.type';

export interface ISqlAuditStatusTag {
  status: getSQLAuditRecordsV1FilterSqlAuditStatusEnum;
}

const SqlAuditStatusTag = ({ status }: ISqlAuditStatusTag) => {
  const { t } = useTranslation();
  const color = useMemo(() => {
    const allColor = {
      auditing: 'blue',
      successfully: 'green'
    };
    return allColor[status];
  }, [status]);
  return (
    <>
      <BasicTag size="small" color={color as BasicTagColor}>
        {t(`sqlAudit.list.status.auditStatus.${status}`)}
      </BasicTag>
    </>
  );
};

export default SqlAuditStatusTag;
