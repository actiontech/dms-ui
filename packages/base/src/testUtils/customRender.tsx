import { Dictionary } from '@actiontech/shared/lib/types/common.type';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@mui/system';
import { Provider } from 'react-redux';
import { MemoryRouter, MemoryRouterProps } from 'react-router-dom';
import { storeFactory } from '@actiontech/shared/lib/testUtil/mockRedux';
import lightTheme from '@actiontech/shared/lib/theme/light';
import basePackageTheme from '../theme/light';

export type RenderParams = Parameters<typeof render>;

const themeData = {
  ...lightTheme,
  ...basePackageTheme
};

export const renderWithReduxAndTheme = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(ui, {
    wrapper: ({ children }) => (
      <ThemeProvider theme={themeData}>
        <Provider store={storeFactory(initStore)}>{children}</Provider>
      </ThemeProvider>
    ),
    ...option
  });
};

export const superRender = (
  ...[ui, option, otherProps]: [
    ...RenderParams,
    {
      routerProps?: MemoryRouterProps;
      initStore?: any;
    }?
  ]
) => {
  const renderReturn = render(ui, {
    wrapper: ({ children }) => {
      return (
        <Provider store={storeFactory(otherProps?.initStore)}>
          <MemoryRouter {...otherProps?.routerProps}>
            <ThemeProvider theme={themeData}>{children}</ThemeProvider>
          </MemoryRouter>
        </Provider>
      );
    },
    ...option
  });
  return renderReturn;
};
