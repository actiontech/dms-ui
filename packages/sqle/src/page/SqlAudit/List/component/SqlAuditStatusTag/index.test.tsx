import SqlAuditStatusTag from '.';
import { renderWithReduxAndTheme } from '@actiontech/shared/lib/testUtil/customRender';
import { getSQLAuditRecordsV1FilterSqlAuditStatusEnum } from '@actiontech/shared/lib/api/sqle/service/sql_audit_record/index.enum';

describe('sqle/SqlAudit/SqlAuditStatusTag', () => {
  it('should match snap shot when status is auditing', () => {
    const { baseElement } = renderWithReduxAndTheme(
      <SqlAuditStatusTag
        status={getSQLAuditRecordsV1FilterSqlAuditStatusEnum.auditing}
      />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('should match snap shot when status is successfully', () => {
    const { baseElement } = renderWithReduxAndTheme(
      <SqlAuditStatusTag
        status={getSQLAuditRecordsV1FilterSqlAuditStatusEnum.successfully}
      />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
