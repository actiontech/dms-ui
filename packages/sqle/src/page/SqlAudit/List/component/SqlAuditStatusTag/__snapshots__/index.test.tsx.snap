// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sqle/SqlAudit/SqlAuditStatusTag should match snap shot when status is auditing 1`] = `
<body>
  <div>
    <span
      class="ant-tag ant-tag-geekblue basic-tag-wrapper basic-small-tag-wrapper css-7o2rh7"
    >
      审核中
    </span>
  </div>
</body>
`;

exports[`sqle/SqlAudit/SqlAuditStatusTag should match snap shot when status is successfully 1`] = `
<body>
  <div>
    <span
      class="ant-tag ant-tag-green basic-tag-wrapper basic-small-tag-wrapper css-1ubynud"
    >
      审核成功
    </span>
  </div>
</body>
`;
